src/form/gig/component.js

```diff
@@ -301,32 +302,76 @@ const repeat = (node, values, repeatIndex) => {
   return node;
 };
 
-const findRepeatTrigger = (node) => {
-  if (node?.servar?.repeated) return node;
+const getTextVariables = (node) => {
+  let textVariables = [];
+
+  const text = node?.properties?.text;
+  if (text) {
+    const match = text.match(TEXT_VARIABLE_PATTERN);
+    if (match) textVariables = match;
+  }
+
+  return textVariables.map((variable) => variable.slice(2, -2));
+};
+
+const getAllTextVariables = (node, variables = []) => {
+  const textVariables = getTextVariables(node);
+  if (textVariables)
+    textVariables.forEach((variable) => variables.push(variable));
```

Comment:
I've not really been ensuring that all functions are pure, but it stuck out to me here that we are pushing directly into the param

Reply:
Yeah likely better to return an array from the recursive function that you merge instead of doing param side effects

Reply:
Yeah, I am pushing to it on purpose. This function is called recursively. The initial array is passed the recursive calls.